File: POO/Python/Encapsulamento.py
class ContaBancaria:
    def __init__(self, saldo_inicial):
        self.__saldo = saldo_inicial  # Atributo privado

    def depositar(self, valor):
        self.__saldo += valor

    def sacar(self, valor):
        if valor <= self.__saldo:
            self.__saldo -= valor
        else:
            print("Saldo insuficiente.")

    def get_saldo(self):
        return self.__saldo

# Explicação: __saldo é um atributo privado 
#   (Python usa dois underlines). O acesso é feito 
#   apenas por métodos depositar, sacar e get_saldo.

File: POO/Java/Encapsulamento.java
package POO.Java;

public class Encapsulamento {
    class ContaBancaria {
        private double saldo; // Protegido: só acessado por métodos
    
        public ContaBancaria(double saldoInicial) {
            this.saldo = saldoInicial;
        }
    
        public void depositar(double valor) {
            saldo += valor;
        }
    
        public void sacar(double valor) {
            if (valor <= saldo) {
                saldo -= valor;
            } else {
                System.out.println("Saldo insuficiente.");
            }
        }
    
        public double getSaldo() {
            return saldo;
        }
    }
    
}

// Explicação: O saldo está protegido (private). 
// O acesso e modificação só podem ser feitos com 
// getSaldo(), depositar() ou sacar() — isso é encapsulamento.